xds/src/main/java/io/grpc/xds/LookasideChannelLb.java

```diff
@@ -0,0 +1,267 @@
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import io.envoyproxy.envoy.api.v2.ClusterLoadAssignment;
+import io.envoyproxy.envoy.api.v2.ClusterLoadAssignment.Policy.DropOverload;
+import io.envoyproxy.envoy.api.v2.endpoint.LocalityLbEndpoints;
+import io.envoyproxy.envoy.type.FractionalPercent;
+import io.envoyproxy.envoy.type.FractionalPercent.DenominatorType;
+import io.grpc.LoadBalancer;
+import io.grpc.LoadBalancerRegistry;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.Status;
+import io.grpc.internal.ExponentialBackoffPolicy;
+import io.grpc.xds.LoadReportClient.LoadReportCallback;
+import io.grpc.xds.LoadReportClientImpl.LoadReportClientFactory;
+import io.grpc.xds.LocalityStore.LocalityStoreImpl;
+import io.grpc.xds.XdsComms.AdsStreamCallback;
+import io.grpc.xds.XdsComms.LbEndpoint;
+import io.grpc.xds.XdsComms.LocalityInfo;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+
+/**
+ * A load balancer that has a lookaside channel. This layer of load balancer creates a channel to
+ * the remote load balancer. LrsClient, LocalityStore and XdsComms are three branches below this
+ * layer, and their implementations are provided by their factories.
+ */
+final class LookasideChannelLb extends LoadBalancer {
+
+  private final ManagedChannel lbChannel;
+  private final LoadReportClient lrsClient;
+  private final AbstractXdsComms xdsComms;
+
+  @VisibleForTesting
+  LookasideChannelLb(
+      Helper helper,
+      final AdsStreamCallback adsCallback,
+      String balancerName,
+      LoadReportClientFactory lrsClientFactory,
+      LoadBalancerRegistry lbRegistry,
+      LocalityStoreFactory localityStoreFactory,
+      XdsCommsFactory xdsCommsFactory) {
+    lbChannel = initLbChannel(helper, balancerName);
+    final LocalityStore localityStore = localityStoreFactory.newLocalityStore(helper, lbRegistry);
+    LoadReportCallback lrsCallback =
+        new LoadReportCallback() {
+          @Override
+          public void onReportResponse(long reportIntervalNano) {
+            localityStore.updateOobMetricsReportInterval(reportIntervalNano);
+          }
+        };
+    lrsClient = lrsClientFactory.createLoadReportClient(
+        lbChannel, helper, new ExponentialBackoffPolicy.Provider(),
+        localityStore.getLoadStatsStore());
+
+    AdsStreamCallback2 adsCallback2 = new AdsStreamCallback2Impl(
+        adsCallback, lrsClient, lrsCallback, localityStore) ;
+    xdsComms = xdsCommsFactory.newXdsComms(lbChannel, adsCallback2);
+
+    // TODO(zdapeng): Maybe executeLater xdsComms.start() to prevent receiving EDS response before
+    // everything has been setup/initialized
+    xdsComms.start();
+  }
+
+  private static int rateInMillion(FractionalPercent fractionalPercent) {
```

Comment:
May want to rename this method.

Reply:
Suggestions? dropsPerMillionRequests?

Reply:
Hmm... That sounds even worse. 

`perMillion`? (Isn't better...)
It's hard to come up one. Maybe just leave `rateInMillion` for now.